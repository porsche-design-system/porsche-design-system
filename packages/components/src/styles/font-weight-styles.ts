import type { TextWeight } from '../types';
import { fontWeightRegular, fontWeightSemiBold, fontWeightBold } from '@porsche-design-system/utilities-v2';
import type { TextWeightDeprecated } from '../components/text/text-weight';

const fontWeightMap: Record<Exclude<TextWeight, TextWeightDeprecated>, number> = {
  regular: fontWeightRegular,
  'semi-bold': fontWeightSemiBold,
  bold: fontWeightBold,
};

export const getFontWeight = (weight: Exclude<TextWeight, TextWeightDeprecated>): number => {
  return fontWeightMap[weight];
};
