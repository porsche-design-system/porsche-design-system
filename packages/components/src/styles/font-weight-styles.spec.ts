import { getFontWeight } from './font-weight-styles';

describe('getFontWeight()', () => {
  it.each<Parameters<typeof getFontWeight>>([['regular'], ['semi-bold'], ['bold']])(
    'should return correct numeric value for weight: %s',
    (...args) => {
      expect(getFontWeight(...args)).toMatchSnapshot();
    }
  );
});
