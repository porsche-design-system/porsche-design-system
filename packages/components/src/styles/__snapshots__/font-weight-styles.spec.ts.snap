// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getFontWeight() should return correct numeric value for weight: bold 1`] = `700`;

exports[`getFontWeight() should return correct numeric value for weight: regular 1`] = `400`;

exports[`getFontWeight() should return correct numeric value for weight: semi-bold 1`] = `600`;
